apps/storybook/src/Text.stories.tsx

```diff
@@ -28,8 +32,18 @@ const TextStory: Story<TextProps> = (args) => {
   return <Text {...rest}>{children}</Text>;
 };
 
-export const Basic: Story<TextProps> = TextStory.bind({});
-Basic.args = {} as TextProps;
+export const Body: Story<TextProps> = TextStory.bind({});
+Body.args = { variant: 'body' };
+export const Small: Story<TextProps> = TextStory.bind({});
+Small.args = { variant: 'small' };
+export const Leading: Story<TextProps> = TextStory.bind({});
+Leading.args = { variant: 'leading' };
+export const Subheading: Story<TextProps> = TextStory.bind({});
+Subheading.args = { variant: 'subheading' };
+export const Title: Story<TextProps> = TextStory.bind({});
+Title.args = { variant: 'title' };
+export const Headline: Story<TextProps> = TextStory.bind({});
+Headline.args = { variant: 'headline' };
```

Comment:
These examples need to also carry over the `as` prop to be the same as the element that was previously being rendered by these individual components. e.g. `<Headline>` becomes `<Text variant='headline' as ='h1'>`.

Also storybook is showing errors for these.

Reply:
`<Small>` should use `as: 'small'`

and this is the error in visual tests:
![](https://user-images.githubusercontent.com/9084735/200344333-e70c2d87-095e-4aff-8399-8cb28058443e.png)

Reply:
Updated. Should `leading` be h4 or h3? `Leading` component is h3 but jsdoc in `Text` says h4

Reply:
Oh wow, now I don't know if the jsdoc is wrong or if the component is wrong. See, this is why I think it was a good idea to deprecate these old components! They abstract the heading levels to the point where it will hurt accessibility.

I would say leave the component as `h3`. This will make it so that we don't break heading levels for existing users.

I'm also thinking we should emphasize that the user should decide what to put in the `as` prop depending on the context of their app, rather than mindlessly replacing `<Leading>` with `<Text variant='leading' as='h4'>`.

Reply:
Changed h4 to h3.

> I'm also thinking we should emphasize that the user should decide what to put in the `as` prop depending on the context of their app, rather than mindlessly replacing `<Leading>` with `<Text variant='leading' as='h4'>`.

Do you have in mind jsdoc or maybe doc website? I think jsdoc might be too limiting when explaining users how to use polymorphic components.

Reply:
> And what about margin, which was always there with Typography components, but was not with `Text` ? (sorry if it is added now)

I think spacing should also be controlled by the user. Having margins built into the components usually hurts more than helps. We've had complaints in the past about this margin: https://github.com/iTwin/iTwinUI-react/issues/60

> Changed h4 to h3.

I think we could keep the story as h4, since it is a new addition that does not need backwards compatibility.

> Do you have in mind jsdoc or maybe doc website? I think jsdoc might be too limiting when explaining users how to use polymorphic components.

Maybe we can add one sentence in the jsdoc and link to a resource about heading levels? Docs site is not published yet, and we want this deprecation to be released as part of v2.

Reply:
Right, but thinkin from the breaking changes and how everyone relied on this one, it will be huge.
Plus, having our margins set, we can guarantee consistency between apps. 
All the heading elements having margin makes sense to me (you always need some kind of a spacing between that and content).
Plus, if we still follow the native elements and keeping ours as close as those, native `h` elements do have margins which we do reset because of our css. So I think if we just deliver `Text` element and actually reset the native elements margin, we do more bad than good.
IMO
@FlyersPh9 anything on this one from the UI standards perspective? Or we have fully removed all spacings from our typography elements in css too?

Reply:
> Right, but thinkin from the breaking changes and how everyone relied on this one, it will be huge.

That's why we are not removing the old components completely. They can continue using the deprecated components.

> Plus, having our margins set, we can guarantee consistency between apps.
All the heading elements having margin makes sense to me (you always need some kind of a spacing between that and content).

We don't know where exactly each heading will be used. The same heading can be used in two completely different contexts where they need different spacing.

And even if they need the same spacing, the dev might want to apply that spacing through flex/grid `gap` rather than margin.

---

Here's an example where `Subheading` is used without spacing:
![](https://user-images.githubusercontent.com/9084735/200364735-0e3761d9-bc80-44e1-9abd-a315a62eae3f.png)

Here's an example where `Title` is used without spacing:
![](https://user-images.githubusercontent.com/9084735/200365498-ef6b59ef-f4cc-493c-a8d2-f9f672b38130.png)

Reply:
Mayank captured 2 good examples of when we'd not want to include that spacing.  The spacing is useful for when building pages like a blog post or documentation page, but the spacing gets in the way when trying to build parts of a component.

Reply:
Might not be a good idea, but what if we provide some type of prop for toggling whether to use default text margins or not in the `ThemeProvider`? Just a thought, not sure if it is good, maybe overkill.

That way devs can use the default margins (we provide consistency across iTwinUI websites) or override the behavior in the ThemeProvider just _once_ to handle _all_ the text spacing themselves (we also provide easy customizability).

> we are not removing the old components completely. They can continue using the deprecated components.

Also, that way users don't need to use deprecated Text components to use our default text spacing values. Because we may remove the deprecated Text components in v3.